resteasy-jaxrs/src/main/java/org/jboss/resteasy/spi/ResteasyProviderFactory.java

```diff
@@ -2561,18 +2561,7 @@ public RuntimeType getRuntimeType()
    @Override
    public boolean isEnabled(Feature feature)
    {
-      Collection<Feature> enabled = getEnabledFeatures();
-      //logger.info("********* isEnabled(Feature): " + feature.getClass().getName() + " # enabled: " + enabled.size());
-      for (Feature f : enabled)
-      {
-         //logger.info("  looking at: " + f.getClass());
-         if (f == feature)
-         {
-            //logger.info("   found: " + f.getClass().getName());
-            return true;
-         }
-      }
-      return false;
+      return getEnabledFeatures().contains(feature);

```

Comment:
@NicoNes, hmmm. You know, I think there used to be a bug, and I think there still is. Consider
```
@Test
public void testEnabled() throws Exception {
   Feature f1 = new TestFeature();
   Feature f2 = new TestFeature();
   ResteasyProviderFactory factory = ResteasyProviderFactory.getInstance();
   factory.register(f1);
   boolean b = factory.isEnabled(f2);
   System.out.println("b: " + b);
}
```
The answer should be "true", right? But it will be false because the code, both old and new, is testing == instead of equals().

Note that the javadoc says
```
Method returns true only in case an instance equal to the feature instance is already present
```
I guess the use of the word "equal" is somewhat ambiguous, but I think it means equals(). What do you think?

-Ron

Reply:
Hi @ronsigal ,

I agree with you, the word "equal" means "equals()" for me too and that's why I changed the code to use `getEnabledFeatures().contains(feature)`.
The contains() method will first check the equality using == and then using equals() if needed.
That's what I'm testing at line https://github.com/resteasy/Resteasy/pull/1666/files#diff-60eb6107f10fea897504eba5e2a8ff10R216.
Am I missing something ?

Reply:
@NicoNes, forget what I just said. Collection.contains() used equals(). Sorry.

The update looks good.

Reply:
@NicoNes, ah, I missed your response.

Reply:
:+1: